### PR TITLE
implement Spring Security

### PR DESCRIPTION
**Pull request by laziz511**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
     <properties>
         <java.version>17</java.version>
-        <spring.version>3.2.1</spring.version>
+        <spring.version>3.0.5</spring.version>
```
Comment (sherzodbeek) [inline pom.xml:23]: What is the purpose of decreasing spring version?

File: web/src/main/java/com/epam/esm/web/controller/OrderController.java [hunk 2]
```diff
@@ -30,17 +32,20 @@ public CollectionModel<OrderModel> getOrders(
     }
 
     @GetMapping("/{id}")
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
     public OrderModel getOrder(@PathVariable("id") Long id) {
         return orderModelAssembler.toModel(orderService.findById(id));
     }
 
     @PostMapping
+    @PreAuthorize("hasRole('ADMIN')")
```
Comment (sherzodbeek) [inline web/src/main/java/com/epam/esm/web/controller/OrderController.java]: Now there is a user in the app. Users can register, which means they should be able to order for themselves also. Admin should not create orders for users.

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -21,7 +33,10 @@
 @Transactional
```
Comment (sherzodbeek) [inline service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: I would advice not using @Transactional at a class level

File: web/src/main/java/com/epam/esm/web/controller/OrderController.java (hunk 2 above)
Comment (sherzodbeek) [reply web/src/main/java/com/epam/esm/web/controller/OrderController.java]: In addition, users should be able to see only its own orders

File: pom.xml (hunk 1 above)
Comment (laziz511) [reply pom.xml:23]: There was a problem in dependencies with Jackson Databind versioning.

File: service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (laziz511) [reply service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java]: ok, thanks. I have moved them from class-level to method-level.

File: web/src/main/java/com/epam/esm/web/controller/OrderController.java (hunk 2 above)
Comment (laziz511) [reply web/src/main/java/com/epam/esm/web/controller/OrderController.java]: fixed

File: web/src/main/java/com/epam/esm/web/controller/OrderController.java [hunk 4]
```diff
@@ -30,23 +33,28 @@ public CollectionModel<OrderModel> getOrders(
...
+    @PreAuthorize("hasAnyRole('USER','ADMIN')")
+    public OrderModel createOrder(@RequestBody @Valid OrderDTO dto, Authentication authentication) {
+
+        Order createdOrder = orderService.create(dto, authentication);
         return orderModelAssembler.toModel(createdOrder);
     }
 
     @GetMapping("/{userId}/user-orders")
+    @PreAuthorize("hasAnyRole('ADMIN', 'USER')")
     public CollectionModel<OrderModel> getOrdersByUserId(
```
Comment (sherzodbeek) [inline web/src/main/java/com/epam/esm/web/controller/OrderController.java:51]: This endpoint is a bit unclear for me. In method arguments, we can see user id and authentication also. Which one is being used to retrieve data? I would advise creating separate endpoints for admin and users also

**Review by sherzodbeek — APPROVED** [review]
LGTM

**Pull request merged**